rust/routee-compass-powertrain/src/routee/prediction/interpolation/interpolation_speed_grade_model.rs

```diff
@@ -33,17 +33,34 @@ impl PredictionModel for InterpolationSpeedGradeModel {
         let grade_value = grade_unit.convert(&grade, &self.grade_unit).as_f64();
 
         // snap incoming speed and grade to the grid
-        let min_speed = self.interpolator.x[0].0;
-        let max_speed = self.interpolator.x[self.interpolator.x.len() - 1].0;
-        let min_grade = self.interpolator.y[0].0;
-        let max_grade = self.interpolator.y[self.interpolator.y.len() - 1].0;
+        let (min_speed, max_speed, min_grade, max_grade) = match &self.interpolator {
+            interp::Interpolator::Interp2D(interp) => (
+                interp.x[0],
+                *interp.x.last().ok_or_else(|| {
+                    TraversalModelError::PredictionModel(
+                        "Could not get last x-value; are x-values empty?".to_string(),
+                    )
+                })?,
+                interp.y[0],
```

Comment:
Hey guys one least question, sorry for being a bottleneck here. Using the index operator here will cause a panic if the response is not as expected. Shouldn't we be using .get instead?

Reply:
Yeah that's a good point. Ideally we would never get to this point if the interpolation grid was empty but it's probably good practice to use get instead (unless that has a performance cost associated with it). Perhaps it would be good enough to just add an additional check into the `validate` function that checks to make sure the grid data always has at least two points and give an instructive error at that point.